integration/setup/setup.go

```diff
@@ -161,9 +166,12 @@ func SetupWithOpts(tb testtb.TB, opts *SetupOpts) *SetupResult {
 	// register cleanup function after setupListener registers its own to preserve full logs
 	tb.Cleanup(cancel)
 
-	collection := setupCollection(tb, setupCtx, client, opts)
+	if opts.SetupUser {
+		// user is created before the collection so that user can run collection cleanup
```

Comment:
@henvic creating user before collection and using authenticated client works :+1: